backend_new/src/utilities/deep-find.ts

```diff
@@ -0,0 +1,13 @@
+export function deepFind(obj: Record<string, any>, path: string) {
+  const paths = path.split('.');
+  let current = obj;
+
+  for (let i = 0; i < paths.length; ++i) {
+    if (current[paths[i]] == undefined) {
+      return undefined;
+    } else {
+      current = current[paths[i]];
+    }
+  }
```

Comment:
I know this was copied from the old code base, but this could be just slightly simplified
```
  paths.forEach((path) => {
    if (current[path] == undefined) {
      return undefined;
    }
    current = current[path];
  });
  ```

Reply:
I had to rework this to for..of loop instead but looks cleaner

nice call out!